pkg/innerring/innerring.go

```diff
@@ -418,6 +418,15 @@ func New(ctx context.Context, log *zap.Logger, cfg *config.Config, errChan chan<
 
 	serveMetrics(server, cfg)
 
+	// TODO: remove deprecated 'fschain_autodeploy' config option in future release
+	if cfg.IsSet("fschain_autodeploy") && cfg.IsSet("fschain.disable_autodeploy") {
+		return nil, fmt.Errorf("'fschain_autodeploy' and 'fschain.disable_autodeploy' set simultaneously")
+	}
+	if cfg.IsSet("fschain_autodeploy") {
```

Comment:
may be merged with the previous `if`. Not a big deal taking into account further removal of both blocks